flathunter/abstract_crawler.py

```diff
@@ -3,50 +3,25 @@
 import logging
 import random
 import requests
+from itertools import cycle
 from bs4 import BeautifulSoup
-
+from random_user_agent.user_agent import UserAgent
+from flathunter import proxies
 
 class Crawler:
     """Defines the Crawler interface"""
 
     __log__ = logging.getLogger('flathunt')
     URL_PATTERN = None
 
-    USER_AGENTS = [
-        # Chrome
-        ("Mozilla/5.0 (Windows NT 10.0; Win64; x64) "
-         "AppleWebKit/537.36 (KHTML, like Gecko)"
-         "Chrome/74.0.3729.169 Safari/537.36"),
-        ("Mozilla/5.0 (Windows NT 10.0; Win64; x64) "
-         "AppleWebKit/537.36 (KHTML, like Gecko) "
-         "Chrome/80.0.3987.149 Safari/537.36"),
-        # Edge
-        ("Mozilla/5.0 (Windows NT 10.0; Win64; x64) "
-         "AppleWebKit/537.36 (KHTML, like Gecko) "
-         "Chrome/51.0.2704.79 Safari/537.36 Edge/14.14393"),
-        ("Mozilla/5.0 (Windows NT 10.0; Win64; x64) "
-         "AppleWebKit/537.36 (KHTML, like Gecko) "
-         "Chrome/70.0.3538.102 Safari/537.36 Edge/18.18363"),
-        # Firefox
-        ("Mozilla/5.0 (Windows NT 6.1; WOW64; rv:40.0) "
-         "Gecko/20100101 Firefox/40.1"),
-        ("Mozilla/5.0 (Windows NT 6.1; WOW64; rv:54.0) "
-         "Gecko/20100101 Firefox/54.0"),
-        # Safari
-        ("Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) "
-         "AppleWebKit/605.1.15 (KHTML, like Gecko) Version/11.1.2 "
-         "Safari/605.1.15"),
-        ("Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) "
-         "AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.5 "
-         "Safari/605.1.15")
-    ]
+    user_agent_rotator = UserAgent()
```

Comment:
This is cool - much tidier to use a library that to have the constants hard-coded